### PR TITLE
Task 30: FE: Implement global confirmation dialog

### PR DESCRIPTION
**Pull request by di-kinosian**

### CODE REVIEW

File: src/components/Dialog/context/ConfirmationDialogContext.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    </ConfirmationDialogContext.Provider>
+  );
+};
+
+export const useConfirmationDialog = (): ConfirmationDialogContextType => {
+  const context = useContext(ConfirmationDialogContext);
+  if (!context) {
+    throw new Error("useDialog must be used within a DialogProvider");
+  }
+  return context;
```
Comment (VYaniv) [inline src/components/Dialog/context/ConfirmationDialogContext.tsx:67]: Empty lines

File: src/components/Layout/MainHeader/MainHeader.tsx [hunk 2]
```diff
@@ -40,7 +46,12 @@ export const MainHeader = () => {
       )}
     >
       <div className="h-16 flex justify-between items-center max-w-[1200px] w-full mx-auto">
-        <img src="/freshnesecom.svg" alt="" />
+        <img
+          src="/freshnesecom.svg"
+          alt=""
+          onClick={() => goToMainPage()}
```
Comment (VYaniv) [inline src/components/Layout/MainHeader/MainHeader.tsx:52]: Please fix the previous comments and update with develop this PR

File: src/components/Dialog/context/ConfirmationDialogContext.tsx (hunk 1 above)
Comment (di-kinosian) [reply src/components/Dialog/context/ConfirmationDialogContext.tsx]: done

File: src/components/Layout/MainHeader/MainHeader.tsx (hunk 2 above)
Comment (di-kinosian) [reply src/components/Layout/MainHeader/MainHeader.tsx]: done

**Review by VYaniv — APPROVED** [review]

**Pull request merged**